CMakeLists.txt

```diff
@@ -308,6 +308,8 @@ if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
   # Drop unreachable code and data.
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-dead_strip,-dead_strip_dylibs")
 
+  check_and_add_flag(NO_OBJC_EXCEPTIONS -fno-objc-exceptions)
```

Comment:
This change probably warrants a comment, so it doesn't accidentally get removed in the future or anything.